src/ln/ln.rs

```diff
@@ -143,39 +143,45 @@ pub fn uumain(args: Vec<String>) -> i32 {
 }
 
 fn exec(files: &[PathBuf], settings: &Settings) -> i32 {
-    match settings.target_dir {
-        Some(ref name) => return link_files_in_dir(files, &PathBuf::from(name), &settings),
-        None => {}
+    if files.len() == 0 {
+        show_error!("missing file operand\nTry '{} --help' for more information.", NAME);
+        return 1;
+    }
+
+    // Handle cases where we create links in a directory first.
+    if let Some(ref name) = settings.target_dir {
+        // 4th form: a directory is specified by -t.
+        return link_files_in_dir(files, &PathBuf::from(name), &settings);
+    }
+    if !settings.no_target_dir {
+        if files.len() == 1 {
+            // 2nd form: the target directory is the current directory.
+            return link_files_in_dir(files, &PathBuf::from("."), &settings);
```

Comment:
Yeah, I suspected it, too. Actually, I've already tried env::current_dir() before but it didn't work. Now I've debugged this on AWS and found Windows doesn't like slashes in symlink targets. It does create a symlink but reading info from it seems to fail.

I've updated the test and now it passes. Another fix would be replacing slashes to backslashes in ln.rs. I'm not sure whether is better as I'm not good at Windows. I'm guessing the current behavior is OK. IIUC symlink is a second-class citizen on Windows anyway.